server/chat-commands/info.js

```diff
@@ -2398,33 +2398,29 @@ const commands = {
 	code(target, room, user) {
 		if (!target) return this.parse('/help code');
 		if (!this.canTalk()) return;
-		if (target.startsWith('\n')) target = target.slice(1);
 		if (target.length >= 8192) return this.errorReply("Your code must be under 8192 characters long!");
-		const separator = '\n';
-		if (target.includes(separator) || target.length > 150) {
-			const params = target.split(separator);
-			let output = [];
-			let cutoff = 3;
-			for (const param of params) {
-				if (output.length < 2 && param.length > 80) cutoff = 2;
-				output.push(Chat.escapeHTML(param));
-			}
-			let code;
-			if (output.length > cutoff) {
-				code = `<div class="chat"><details class="readmore code" style="white-space: pre-wrap; display: table; tab-size: 3"><summary>${output.slice(0, cutoff).join('<br />')}</summary>${output.slice(cutoff).join('<br />')}</details></div>`;
-			} else {
-				code = `<div class="chat"><code style="white-space: pre-wrap; display: table; tab-size: 3">${output.join('<br />')}</code></div>`;
-			}
 
-			if (!this.canBroadcast(true, '!code')) return;
-			if (this.broadcastMessage && !this.can('broadcast', null, room)) return false;
-
-			if (!this.runBroadcast(true, '!code')) return;
+		const params = target.split('\n');
```

Comment:
Here is my compromise for minimum length.

```suggestion
		const params = target.split('\n');
		if (!params[0]) params.unshift();
		if (!params[params.length]) params.pop();
		if (params.length === 1 && params[0].length < 80 && !target.includes('```')) {
			return `\`\`\`${params[0]}\`\`\``;
		}
```

Reply:
I'm mostly alright with this. I think something like this would work better for the first half:
```javascript
while (!params[0]) params.unshift();
while (!params[params.length - 1]) params.pop();
```
What's the `this.message.startsWith('/')` check in the second half for?

Reply:
I'm not using `while` because I want to support people intentionally adding leading/trailing linebreaks

Reply:
The `startsWith('/')` check is because `/code` should not degrade to broadcasting publicly, only `!code`.